napari/layers/image/image.py

```diff
@@ -297,13 +297,16 @@ def __init__(
         self._experimental_slicing_plane = SlicingPlane(
             thickness=1, enabled=False
         )
+        # Whether to calculate clims on the next set_view_slice
+        self._should_calc_clims = False
         if contrast_limits is None:
             if not isinstance(data, np.ndarray):
                 dtype = getattr(data, 'dtype', None)
                 if np.issubdtype(normalize_dtype(dtype), np.integer):
                     self.contrast_limits_range = get_dtype_limits(dtype)
                 else:
                     self.contrast_limits_range = (0, 1)
+                self._should_calc_clims = True
```

Comment:
I would actually exclude np.uint8, and maybe even np.uint16 from this. It's a common issue that the first plane is actually completely black, and this will cause bigger issues than it solves...

Reply:
excluded uint8.  don't really want to do uint16, (since that was actually the case that made me want to open this issue! :)

Reply:
that's fine with me